kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+band_paths
+
+# Open & clip the data - a single band using rioxarray
+band = rxr.open_rasterio(band_paths[0], masked=True).rio.clip(crop_bound.geometry,
+                                                              from_disk=True).squeeze()
+
+# Specify the valid range of values for landsat
+valid_range = (0, 10000)
+
+if valid_range:
+    mask = ((band < valid_range[0]) | (band > valid_range[1]))
+    band = band.where(~xr.where(mask, True, False))
+
+band
+
+# Open and clean a single band using a function
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+# %%timeit
+
+# Open and clean a single band
+band_4 = open_clean_bands(band_path=band_paths[0],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+# Open and clean a single band
+band_5 = open_clean_bands(band_path=band_paths[1],
+                          crop_extent=crop_bound,
+                          # The range of valid values for landsat can include negative values
+                          # for this week let's stick with 0-10000
+                          valid_range=(0, 10000))
+
+#  Then  calculate NDVI
+ndvi = ((band_5-band_4) / (band_5+band_4))
+
+ep.plot_bands(ndvi,
+              cmap='PiYG',
+              vmin=-1)
+plt.show()
+
+
+# In[7]:
+
+
+# Create dataframe of mean NDVI in this cell using the functions created above
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+
+# Call the dataframe at the end of the cell so the tests run on it!
+# Be sure that the date column is an index of type date
+# HINT: the time series lessons may help you remember how to do this!
+
+# YOUR CODE HERE
+
+path = os.path.join("ndvi-automation", "sites")
+
+# Get a list of both site directories (We will talk more about automation next week)
+sites = glob(path + "/*/")
+
+# Get the site name
+site_name = os.path.basename(os.path.normpath(sites[0]))
+
+# Open up the shapefile for clipping your landsat data to the study area
+vector_dir = os.path.join(sites[0],
+                          "vector")
+
+# Open crop boundary
+site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+
+# Import tif
+harv_path = sorted(glob(os.path.join("ndvi-automation",
+                        "sites",
+                        "HARV",
+                         "landsat-crop",
+                         "LC080130302017031701T1-SC20181023151837",
+                         "*band*[4-5].tif")))
+
```

Comment:
This section of code has in-line comments, which is inconsistent with PEP8 standards, but I see it enough in code I've used, that I assume it's common practice.

Reply:
Yes, I agree. I generally don't do in-line, but for this particular block of code, I found it helpful to contextualize each line to make sure I was being efficient, organized, and understanding what I was doing. (again, brand new coder here!) 

I was conflicted on whether to move these above when I was finished in a summarized way (or remove them) and be in PEP8 standards, or to leave them there to demonstrate to the grader (and my future self if I reference this assignment again) that I had worked through each line to understand each piece that I wrote.

Another case where I need to make a note to self that some of my practices will be problematic outside of academia.

Thank you!